polkadot/node/core/prospective-parachains/src/fragment_chain/mod.rs

```diff
@@ -313,16 +358,81 @@ pub(crate) struct CandidateEntry {
 }
 
 impl CandidateEntry {
+	/// Create a new seconded candidate entry.
+	pub fn new_seconded(
+		candidate_hash: CandidateHash,
+		candidate: CommittedCandidateReceipt,
+		persisted_validation_data: PersistedValidationData,
+	) -> Result<Self, CandidateEntryError> {
```

Comment:
There are multiple cases where this function could fail. I think it's a good practice to just have the user just look at the possible Error variants, rather than mentioning them in doc comments (which can easily become outdated)